neuro_san/internals/run_context/langchain/llms/standard_langchain_llm_factory.py

```diff
@@ -178,27 +174,38 @@ def create_llm_resources_with_client(self, config: Dict[str, Any],
                     "include_usage": True
                 }
             }
-            openai_api_key: str = self.get_value_or_env(config, "openai_api_key", "AZURE_OPENAI_API_KEY")
-            if openai_api_key is None:
-                openai_api_key = self.get_value_or_env(config, "openai_api_key", "OPENAI_API_KEY")
 
             # AzureChatOpenAI just happens to come with langchain_openai
             # pylint: disable=invalid-name
             AzureChatOpenAI = resolver.resolve_class_in_module("AzureChatOpenAI",
                                                                module_name="langchain_openai.chat_models.azure",
                                                                install_if_missing="langchain-openai")
+
+            # Create the policy object that allows us to manage the model run-time lifecycle
+            client_policy = AzureClientPolicy()
+            async_client: Any = client_policy.create_client(config)
+
+            # Prepare some more complex args
+            openai_api_key: str = self.get_value_or_env(config, "openai_api_key", "AZURE_OPENAI_API_KEY", async_client)
+            if openai_api_key is None:
+                openai_api_key = self.get_value_or_env(config, "openai_api_key", "OPENAI_API_KEY", async_client)
+
             llm = AzureChatOpenAI(
+                async_client=async_client,
                 model_name=model_name,
                 temperature=config.get("temperature"),
+
+                # This next group of params should always be None when we have async_client
                 openai_api_key=openai_api_key,
                 openai_api_base=self.get_value_or_env(config, "openai_api_base",
-                                                      "OPENAI_API_BASE"),
+                                                      "OPENAI_API_BASE", async_client),
                 openai_organization=self.get_value_or_env(config, "openai_organization",
-                                                          "OPENAI_ORG_ID"),
-                openai_proxy=self.get_value_or_env(config, "openai_organization",
-                                                   "OPENAI_PROXY"),
```

Comment:
There was a conflation of OPENAI_ORG_ID and OPENAI_PROXY before. Now these are more properly separated.